kubernetes/linux/main.sh

```diff
@@ -334,32 +334,35 @@ if [ -e "/etc/ama-logs-secret/WSID" ]; then
       fi
 
       if [ $? -ne 0 ]; then
+            registry="mcr.microsoft.com"
+            if  [ $CLOUD_ENVIRONMENT == "usnat" ] || [ $CLOUD_ENVIRONMENT == "ussec" ]; then
+                  registry=$MCR_URL
```

Comment:
Please test the changes.

Reply:
Tested the changes. Adding the scenarios in the description